### PR TITLE
[ALLUXIO-3219] Consolidate report configuration with getConf

### PR DESCRIPTION
**Pull request by apc999**
https://alluxio.atlassian.net/browse/ALLUXIO-3219

**Major changes:**

- move original `report configuration` as `getConf --master` and unify its output with `getConf` (which by default shows the local conf)
- when creating the output of getConf CLI and WebUI, show all built-in conf properties (e.g., `aws.secreteKey`), rather than only keys prefixed with string `"alluxio."`
- remove `USER_HOME` which was treated as a built-in `PropertyKey`. It turns out unnecessary.

screen shot 2018-06-14 at 11 20 15 am

### CODE REVIEW

File: shell/src/main/java/alluxio/cli/GetConf.java [hunk 1]
```diff
@@ -137,59 +147,87 @@ public static int getConf(String... args) {
...
       case 1:
-        if (!PropertyKey.isValid(args[0])) {
+        String key = args[0];
+        if (!PropertyKey.isValid(key)) {
           printHelp(String.format("%s is not a valid configuration key", args[0]));
           return 1;
         }
-        PropertyKey key = PropertyKey.fromString(args[0]);
-        if (!Configuration.isSet(key)) {
+        ConfigProperty property = confMap.get(args[0]);
```
[the remark this answers is not in the document]
Comment (apc999) [reply shell/src/main/java/alluxio/cli/GetConf.java]: done

File: shell/src/main/java/alluxio/cli/GetConf.java [hunk 2]
```diff
@@ -137,59 +147,87 @@ public static int getConf(String... args) {
...
+        if (property == null) {
+          printHelp(String.format("%s is not found", args[0]));
```
[the remark this answers is not in the document]
Comment (apc999) [reply shell/src/main/java/alluxio/cli/GetConf.java]: done

File: shell/src/main/java/alluxio/cli/GetConf.java [hunk 3]
```diff
@@ -45,16 +51,20 @@
       + "Possible unit options include B, KB, MB, GB, TP, PB, MS, S, M, H, D. "
       + "if \"--source\" option is specified, the source of configuration value will be printed.";
 
+  private static final String CLUSTER_OPTION_NAME = "cluster";
   private static final String SOURCE_OPTION_NAME = "source";
   private static final String UNIT_OPTION_NAME = "unit";
+  private static final Option CLUSTER_OPTION =
+      Option.builder().required(false).longOpt(CLUSTER_OPTION_NAME).hasArg(false)
+          .desc("the cluster-default configuration property.").build();
```
[the remark this answers is not in the document]
Comment (apc999) [reply shell/src/main/java/alluxio/cli/GetConf.java]: right. text updated.

File: core/common/src/main/java/alluxio/conf/AlluxioProperties.java [hunk 4]
```diff
@@ -133,7 +133,7 @@ public void merge(Map<?, ?> properties, Source source) {
         // This will register the key as a valid PropertyKey
         // TODO(adit): Do not add properties unrecognized by Ufs extensions when Configuration
         // is made dynamic
-        propertyKey = new PropertyKey.Builder(key).build();
+        propertyKey = new PropertyKey.Builder(key).setIsBuiltIn(false).build();
```
[the remark this answers is not in the document]
Comment (apc999) [reply core/common/src/main/java/alluxio/conf/AlluxioProperties.java]: - yes
- a test added.

File: shell/src/main/java/alluxio/cli/GetConf.java [hunk 5]
```diff
@@ -25,17 +28,20 @@
...
 import java.util.List;
+import java.util.Map;
 
 /**
  * Utility for printing Alluxio configuration.
  */
 public final class GetConf {
-  private static final String USAGE = "USAGE: GetConf [--unit <arg>] [--source] [KEY]\n\n"
+  private static final String USAGE =
+      "USAGE: GetConf [--unit <arg>] [--source] [--cluster] [key]\n\n"
```
[the remark this answers is not in the document]
Comment (apc999) [reply shell/src/main/java/alluxio/cli/GetConf.java]: - added into the option description (which will be shown with help)
- I renamed this flag to `--master` let me know if it is fine

File: shell/src/main/java/alluxio/cli/fsadmin/report/ConfigurationCommand.java [hunk 6]
```diff
@@ -1,57 +0,0 @@
...
-import java.io.IOException;
-import java.io.PrintStream;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.List;
-
-/**
- * Prints server-side configuration check report.
- */
-public class ConfigurationCommand {
```
[the remark this answers is not in the document]
Comment (apc999) [reply shell/src/main/java/alluxio/cli/fsadmin/report/ConfigurationCommand.java]: I could not find a clean way . any suggestion?

File: shell/src/main/java/alluxio/cli/GetConf.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (aaudiber) [reply shell/src/main/java/alluxio/cli/GetConf.java]: `--master` sgtm

File: core/common/src/test/java/alluxio/ConfigurationTest.java [hunk 7]
```diff
@@ -854,4 +854,17 @@ public void getCredentialsDisplayValueIdentical() {
...
+  @Test
+  public void extensionProperty() {
+    // simulate the case a ext key is picked by site property
+    String fakeKeyName = "fake.extension.key";
+    Configuration.merge(ImmutableMap.of(fakeKeyName, "value"), Source.siteProperty("a"));
+    assertFalse(PropertyKey.fromString(fakeKeyName).isBuiltIn());
+    // simulate the case the key is built inside the extension
+    PropertyKey fakeExtensionKey = new PropertyKey.Builder(fakeKeyName).build();
+    String value = Configuration.get(fakeExtensionKey);
+    assertEquals("value", value);
```
Comment (aaudiber) [inline core/common/src/test/java/alluxio/ConfigurationTest.java]: nit: can combine to one line

Comment (apc999) [reply core/common/src/test/java/alluxio/ConfigurationTest.java]: done

**Comment by apc999** [comment]
@aaudiber PTAL

File: shell/src/main/java/alluxio/cli/GetConf.java [hunk 8]
```diff
@@ -127,69 +141,109 @@ public static void printHelp(String message) {
...
+  /**
+   * Implements get configuration.
+   *
+   * @param clientSupplier a functor to return a client of meta master
+   * @param args list of arguments
+   * @return 0 on success, 1 on failures
+   */
+  @VisibleForTesting
+  public static int getConfImpl(
+      @Nullable Supplier<RetryHandlingMetaMasterClient> clientSupplier, String... args) {
```
Comment (aaudiber) [inline shell/src/main/java/alluxio/cli/GetConf.java]: I think we can remove the `@Nullable` here

**Pull request merged**